lib/fits.rb

```diff
@@ -0,0 +1,100 @@
+module Fits
+  class Installer
+    attr_accessor :download_url, :install_dir, :symlink, :verbose
+
+    def initialize(opts={})
+      @download_url = opts.delete(:download_url) || 'http://projects.iq.harvard.edu/files/fits/files/fits-0.8.4_0.zip'
+      @install_dir = opts.delete(:install_dir) || Dir.pwd
+      @symlink = opts.delete(:symlink) || 'fits'
+      @verbose = opts.delete(:verbose) || true
+      @orig_dir = Dir.pwd
+    end
+
+
+    def install
+      remove_all!
+      download
+      unzip
+      make_executable
+      make_symlink
+      remove_zip_file
+      Dir.chdir(@orig_dir)
+    end
+
+    def remove_fits!
+      puts_around("Removing #{unzipped_dir}") { FileUtils.rm_r(unzipped_dir, secure: true) if Dir.exists? unzipped_dir }
+    end
+
+    def remove_zip_file
+      within_dir(install_dir) do
+        puts_around("Removing #{File.expand_path(zipped_filename)}") { FileUtils.rm_r(zipped_filename, secure: true) if File.exists? zipped_filename }
+      end
+    end
+
+    def remove_all!
+      remove_zip_file
+      remove_fits!
+    end
+
+    private
+
+    def within_dir(dir)
+      orig_dir = Dir.pwd
+      Dir.chdir(dir)
+      yield if block_given?
+      Dir.chdir(orig_dir)
```

Comment:
@afred : actually, you can pass a block to chdir. We don't need to do this by hand. http://ruby-doc.org/core-2.2.0/Dir.html#method-c-chdir